src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -711,6 +715,26 @@ private void DowngradeLockFileIfNeeded(LockFile lockFile)
             }
         }
 
+        private async Task UpdateLastAccessTime(CacheFile cacheFile)
```

Comment:
One way of reducing the perf hit here would be to update is once per operation, rather than once per project. 

Packages are commonly shared among projects, so it might lead to some good savings. 

I think it could be something that you test on the side and post the perf results for comparison.

Reply:
> One way of reducing the perf hit here would be to update is once per operation, rather than once per project.

Sorry, but I don't understand what you mean. Could you clarify what you mean by 'operation' please?

Reply:
Sure. 

Restore always need to run on a per project level.
Even when projects have project references to each other, the restores there can happen independently.

When you run restore on the commandline, or in VS, multiple restore for different projects are kicked off concurrently.
While the graph generation is fairly independent, things such as making http calls to the sources, package installation, package enumeration etc, are usually cached. 

So if you have 2 projects, that both need `Newtonsoft.Json`, the same http calls will be reused, the package installation will happen only once etc. 

Given that most of the time, the global packages folder across projects is shared, you could do the timestamp update only once. 

For example, Newtonsoft.Json is a package that's a part of ~75% of NuGet's projects. There's really no benefit to updating the timestamp 75 times. 

You consider testing the perf hit if you try to ensure these updates happen only once. 


An easy and recent example is the LockFileBuilderCache.